packages/sitecore-jss-nextjs/src/services/graphql-sitemap-service.ts

```diff
@@ -1,258 +1,259 @@
+type SitePageQueryResult = {
   search: {
-    results: { url: { path: string } }[];
+    pageInfo: {
+      endCursor: string;
+      hasNext: boolean;
+    };
+    results: {
+      url: {
+        path: string;
+      };
+    }[];
   };
 };
 
+/**
+ * Fetch the list of site pages using  Sitecore's GraphQL API. This list is used for SSG and Export functionality.
+ * Note: Uses graphql-request as the default library for fetching graphql data (@see GraphQLRequestClient).
+ */
 export class GraphQLSitemapService {
   /**
-   * Provides ability to fetch sitemap from graphql endpoint.
-   * Sitemap can be used for SSG/Export
-   * @param {GraphQLSitemapServiceConfig} config graphql sitemap service config
+   * Creates an instance of graphQL sitemap service with the provided options
+   * @param {GraphQLSitemapServiceConfig} options instance
    */
-  constructor(private config: GraphQLSitemapServiceConfig) {}
+  constructor(public options: GraphQLSitemapServiceConfig) {
+    this.options.pageSize = this.options.pageSize ?? DEFAULTS.pageSize;
+  }
 
   /**
    * Fetch sitemap which could be used for generation of static pages during `next export`.
    * The `locale` parameter will be used in the item query, but since i18n is not supported,
-   * the output paths will not include a `locale` property.
-   * @param {string} locale locale which application supports
+   * the output paths will not include a `language` property.
+   * @param {string} locale which application supports
    * @param {string} rootItemPath root item path
-   * @param {Function} [formatSearchQuery] override default search query
-   *
-   * @default
-   * Search query
-   * search(
-   *    where: {
-   *      AND:[
-   *        {
-   *           name:"_path",
-   *          value:"${rootItemId.toLowerCase()}"
-   *        },
-   *        {
-   *          name:"_language",
-   *          value:"${locale}"
-   *        },
-   *        {
-   *          name:"_hasLayout",
-   *          value :"true"
-   *        }
-   *      ]
-   *    }
-   *  )
+   * @throws {RangeError} if a valid root item ID cannot be determined from the specified root item path.
+   * @throws {RangeError} if the specified list of locales is empty or contains invalid values.
    */
-  async fetchExportSitemap(
-    locale: string,
-    rootItemPath: string,
-    formatSearchQuery?: (rootItemId: string, locale: string) => string
-  ): Promise<StaticPath[]> {
+  async fetchExportSitemap(locale: string, rootItemPath: string): Promise<StaticPath[]> {
     const formatPath = (path: string[]) => ({
       params: {
         path,
       },
     });
 
-    return this.fetch([locale], rootItemPath, formatPath, formatSearchQuery);
+    return this.fetchSitemap([locale], rootItemPath, formatPath);
   }
 
   /**
    * Fetch sitemap which could be used for generation of static pages using SSG mode
-   * @param {string[]} locales locales which application supports
+   * @param {string[]} locales which application supports
```

Comment:
@anastasiya29 First word - it's argument name, and next words - it's description, previously it was correct

Reply:
👍